openslides_backend/action/actions/topic/import_.py

```diff
@@ -0,0 +1,88 @@
+from typing import Any, Dict
+
+from ....models.models import ActionWorker
+from ....permissions.permissions import Permissions
+from ....shared.exceptions import ActionException
+from ....shared.interfaces.event import Event, EventType
+from ....shared.interfaces.write_request import WriteRequest
+from ....shared.patterns import fqid_from_collection_and_id
+from ....shared.schema import required_id_schema
+from ...action import Action
+from ...util.default_schema import DefaultSchema
+from ...util.register import register_action
+from .create import TopicCreate
+from .mixins import DuplicateCheckMixin
+
+
+@register_action("topic.import")
+class TopicImport(DuplicateCheckMixin, Action):
+    """
+    Action to import a result from the action_worker.
+    """
+
+    model = ActionWorker()
+    schema = DefaultSchema(ActionWorker()).get_default_schema(
+        additional_required_fields={
+            "id": required_id_schema,
+            "command": {"type": "string", "enum": ["import", "abort"]},
+        }
+    )
+    permission = Permissions.AgendaItem.CAN_MANAGE
+
+    def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+        store_id = instance["id"]
+        command = instance["command"]
+        if command == "import":
+            meeting_id = self.get_meeting_id(instance)
+            self.init_duplicate_set(meeting_id)
+            worker = self.datastore.get(
+                fqid_from_collection_and_id("action_worker", store_id), ["result"]
+            )
+            action_payload = [
+                entry["data"]
+                for entry in worker.get("result", [])
+                if entry["status"] == "new"
+                and not self.check_for_duplicate(entry["data"]["title"])
+            ]
+            self.execute_other_action(TopicCreate, action_payload)
+
+        # overwrite, because it is imported or aborted.
+        self.datastore.write_action_worker(
+            WriteRequest(
+                events=[
+                    Event(
+                        type=EventType.Update,
+                        fqid=fqid_from_collection_and_id("action_worker", store_id),
+                        fields={
+                            "id": store_id,
+                            "result": None,
+                        },
+                    )
+                ],
+                user_id=self.user_id,
+                locked_fields={},
+            )
+        )
```

Comment:
Remove this: if the import fails for some reason, we need to have this data still available.

Reply:
Mmh, this also handles the abort. Should an abort remove the data or not?

Reply:
As I see in the later comments, we want to remove abort.

Reply:
> Remove this: if the import fails for some reason, we need to have this data still available.

IMO we should remove the data, if the import is successful. The topic import never will be large, but others will.

Reply:
@jsangmeister seems to have another opinion on this. This is a contradiction. @r-peschke  pls discuss this with @jsangmeister .

Reply:
Yes, but we already discussed that a regular (e.g. daily) cleanup of all "old" import data (e.g. older than two weeks) would be the way to go here. The problem is, if we delete it during the action and the datastore request fails because of locked fields or something else, we cannot retry it. A better solution would be to include it in the `on_success` routine, but even that is not perfect because if something else goes wrong during the `import` request, the user has to regenerate the preview first before he can import again. So, for a better UX, we should keep the data here.